api/controllers/blocks.js

```diff
@@ -87,4 +105,104 @@ BlocksController.getBlocks = function(context, next) {
 	});
 };
 
+/**
+ * Get filtered list of blocks (without transactions - BasicBlock).
+ *
+ * @private
+ * @func _list
+ * @param {Object} filter - Conditions to filter with
+ * @param {string} filter.id - Block id
+ * @param {string} filter.generatorPublicKey - Public key of delegate who generates the block
+ * @param {number} filter.numberOfTransactions - Number of transactions
+ * @param {string} filter.previousBlock - Previous block ID
+ * @param {number} filter.height - Block height
+ * @param {number} filter.totalAmount - Total amount of block's transactions
+ * @param {number} filter.totalFee - Block total fees
+ * @param {number} filter.reward - Block reward
+ * @param {number} filter.limit - Limit of blocks to retrieve, default: 100, max: 100
+ * @param {number} filter.offset - Offset from where to start
+ * @param {string} filter.sort - Sort order, default: height:desc
+ * @param {function} cb - Callback function
+ * @returns {function} cb - Callback function from params (through setImmediate)
+ * @returns {Object} cb.err - Error if occurred
+ * @returns {Object} cb.data - List of normalized blocks
+ */
+function _list(filter, cb) {
+	const options = {};
+
+	const filters = {
+		id: filter.id,
+		generatorPublicKey: filter.generatorPublicKey,
+		numberOfTransactions: filter.numberOfTransactions,
+		previousBlockId: filter.previousBlock,
+		height: filter.height,
+		timestamp_gte: filter.fromTimestamp,
+		timestamp_lte: filter.toTimestamp,
+		totalAmount: filter.totalAmount,
+		totalFee: filter.totalFee,
+		reward: filter.reward,
+	};
+
+	Object.keys(filters).forEach(key => {
+		if (!filters[key]) {
+			delete filters[key];
+		}
+	});
+
+	if (!filter.limit) {
+		options.limit = 100;
+	} else {
+		options.limit = Math.abs(filter.limit);
+	}
```

Comment:
```
options.limit = filter.limit ? Math.abs(filter.limit) : 100;
```

Reply:
I think it's a matter of taste, but if you do that @lsilvs you could also move that line up, where the `filters` object initialization is.